Source/Utils/IntervalsParser.cpp

```diff
@@ -116,3 +122,74 @@ int IntervalsParser::localPeriod (const int n) const
 }
 
 bool IntervalsParser::isActivated () const {return m_activated;}
+
+BTDIntervalsParser::BTDIntervalsParser (const std::vector<std::string>& instr_vec)
```

Comment:
Could you maybe add a few brief inline comments in the body of this method, to describe what each part of the implementation does? Just to make it a bit more readable to someone who hasn't written the code.

Reply:
yes, I remember wondering at the time if I would need this, let alone someone else.  Now it is somewhat annotated